### ISSUE TITLE
`tree.mods.is(Flags.Synthetic)` is inconsistent with `tree.symbol.is(Flags.Synthetic)` past the `posttyper` phase

### ISSUE THREAD

**Issue by mrdziuban** [lampepfl/dotty#18343]
## Compiler version

3.3.0, also happens with the latest 3.3.1-RC4

## Minimized code

https://github.com/mrdziuban/dotty-synthetic-defdef-issue

This consists of two things:

1. A compiler plugin to print out the result of `tree.mods.is(Flags.Synthetic)` and `tree.symbol.is(Flags.Synthetic)` for any `DefDef` it encounters -- https://github.com/mrdziuban/dotty-synthetic-defdef-issue/blob/main/plugin/src/main/scala/PrintSyntheticPlugin.scala#L18-L25
2. A simple test file that defines a `case class Foo()` -- https://github.com/mrdziuban/dotty-synthetic-defdef-issue/blob/main/tests/src/main/scala/PrintSyntheticTest.scala

## Output

The results are inconsistent. `tree.symbol.is` seems to always be correct, but `tree.mods.is` reports that a number of methods are not synthetic. The repo's README has the full output, but here's a sample:

```scala
def <init>:
  mods.is(Flags.Synthetic): false
  symbol.is(Flags.Synthetic): false

def hashCode:
  mods.is(Flags.Synthetic): false
  symbol.is(Flags.Synthetic): true

def equals:
  mods.is(Flags.Synthetic): false
  symbol.is(Flags.Synthetic): true
```

## Expectation

The two approaches for checking whether the `DefDef` is synthetic should be consistent

**Comment by dwijnand** [comment]
I'm not sure I agree with the expectation.  There's no documentation to back me or our expectation, so we need to ask.  But my reading of the code is that tree mods, which is defined for `DefTree` ("Tree defines a new symbol") are a way to define modifiers on what will eventually become the symbol for that tree.  But in some scenarios the compiler may just skip the tree and sets the modifiers on the symbol directly. @odersky what do you think?

I really appreciate all the effort you put in to minimise and document the issue, however.

**Comment by odersky** [comment]
That's correct. If a tree has a symbol, the modifiers of the symbol are what counts. The modifiers of the tree do not matter in that case.

**Comment by mrdziuban** [comment]
Thanks for the info @dwijnand and @odersky. Is/should this be documented anywhere?

**Comment by odersky** [comment]
Yes, would be good to document this on `mods` in `Trees`.
